suite/tests/client-interface/mangle_suspend.c

```diff
@@ -0,0 +1,163 @@
+/* **********************************************************
+ * Copyright (c) 2015-2018 Google, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL GOOGLE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#ifndef ASM_CODE_ONLY /* C code */
+#    include "configure.h"
+#    ifndef UNIX
+#        error UNIX-only
+#    endif
+#    include "tools.h"
+#    include "mangle_suspend-shared.h"
+#    include <setjmp.h>
+#    include <signal.h>
+#    include <pthread.h>
+
+volatile bool test_ready = false;
+volatile bool test_done = false;
+volatile int loop_inc = 1;
+
+/* asm routines */
+void
+test_asm();
+
+static void *
+suspend_thread_routine(void *arg)
+{
+#    ifdef X86_64
+    while (!test_ready) {
+        /* Empty. */
+    }
+    struct timespec sleeptime;
+    sleeptime.tv_sec = 0;
+    sleeptime.tv_nsec = 2000 * 1111;
+    while (!test_done) {
+        asm volatile("mov %0, %%rdx\n\t"
+                     "mov %0, %%rdx\n"
+                     :
+                     : "i"(SUSPEND_VAL_C)
+                     : "rdx");
+        nanosleep(&sleeptime, NULL);
```

Comment:
It is not clear what this thread is for: please add a comment

Reply:
Added comment.